src/lib/pipe_lib/strict_pipe.ml

```diff
@@ -43,7 +43,7 @@ module Reader = struct
 
   let iter ?consumer ?continue_on_error reader ~f =
     enforce_single_reader reader
-      (Pipe.iter reader.reader ?consumer ?continue_on_error ~f)
+      (Pipe.iter_without_pushback reader.reader ?consumer ?continue_on_error ~f)
```

Comment:
You also need to update `fold` to use `Pipe.fold_without_pushback`.

Reply:
okay